x11-misc/yagostatus/yagostatus-0.2.0.ebuild

```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit golang-build
+
+EGO_PN="github.com/burik666/yagostatus"
+DESCRIPTION="Yet Another i3status replacement written in Go"
+HOMEPAGE="https://github.com/burik666/yagostatus"
+LICENSE="GPL-3"
+SLOT="0/${PVR}"
+IUSE="pie"
+
+EGO_VENDOR=(
+	"golang.org/x/net 32a936f github.com/golang/net"
+	"gopkg.in/yaml.v2 5420a8b github.com/go-yaml/yaml"
+)
+
+QA_PRESTRIPPED="usr/bin/yagostatus"
+
+if [[ ${PV} == *9999* ]]; then
+	inherit golang-vcs
+else
+	inherit golang-vcs-snapshot
+	EGIT_COMMIT="v${PV}"
+	SRC_URI="https://${EGO_PN}/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz
+	${EGO_VENDOR_URI}"
+	KEYWORDS="~amd64"
+fi
+
+G="${WORKDIR}/${P}"
+S="${G}/src/${EGO_PN}"
```

Comment:
most likely you don't need re-definition of `${S}` this if you use eclass correctly.